### ISSUE TITLE
Update version on PyPi?

### ISSUE THREAD

**Issue by todbot** [trezor/cython-hidapi#73]
Hi,
The version on PyPi (including the wheels) appears pretty old, from 23 Oct 2017. I see that several updates to `cython-hidapi` have been made since then (including to using libusb/hidapi, thank you! It has even more bug fixes now)

Can I help produce an updated release to PyPi?

**Comment by 0xDP** [comment]
I would appreciate this because hidapi-0.7.99.post21 on macOS has a bug which leads to empty path entries.

This is what I actually get
```
>>> import hid
>>> import re
>>> re.findall('(\'path.+?\'),', str(hid.enumerate()))
["'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''"]
>>> 
```
and this is what I should get
```
>>> import hid
>>> import re
>>> re.findall('(\'path.+?\'),', str(hid.enumerate()))
["'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@1/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80700000/Touch Bar Backlight@80700000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/IOResources/IOHIDResource/IOHIDResourceDeviceUserClient/IOHIDUserDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@2/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/XHC1@14/XHC1@14000000/HS09@14400000/USB to Serial@14400000/IOUSBHostInterface@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@4/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80600000/Touch Bar Display@80600000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80300000/Ambient Light Sensor@80300000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/IOResources/IOHIDResource/IOHIDResourceDeviceUserClient/IOHIDUserDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@3/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80400000/Headset@80400000/HID Relay@0/AppleUserUSBHostHIDDevice'"]
>>> 
```

**Comment by prusnak** [comment]
Uploaded here: https://pypi.org/project/hidapi/0.9.0/#files

Can you please confirm that this works?

**Comment by 0xDP** [comment]
Unfortunately I cannot confirm that this works.
```
% pip list | grep hidapi
hidapi                                 0.9.0
```
The path entries remain empty:
```
>>> import hid
>>> import re
>>> re.findall('(\'path.+?\'),', str(hid.enumerate()))
["'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''", "'path': ''"]
```
It seems to be that the hidapi directory is still the old one from signal 11 and not the one from libusb. At least ```hidapi/mac/hid.c```  still containes the wrong path to the Framework.

**Comment by prusnak** [comment]
I did everything correctly. But unfortunately hidapi still contains this issue in the latest 0.9.0 release.

I just bumped the used hidapi to master and released 0.9.0.post1 package to PyPI: https://pypi.org/project/hidapi/0.9.0.post1/#files

Please check if this works.

**Comment by 0xDP** [comment]
Yep, that works.
```
% pip list | grep hidapi                                  
hidapi                                 0.9.0.post1 
```

```
>>> import hid
>>> import re
>>> re.findall('(\'path.+?\'),', str(hid.enumerate()))
["'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@1/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@1/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@1/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@1/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80700000/Touch Bar Backlight@80700000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@4/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80600000/Touch Bar Display@80600000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80600000/Touch Bar Display@80600000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80600000/Touch Bar Display@80600000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80400000/Headset@80400000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80300000/Ambient Light Sensor@80300000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@3/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/IOResources/IOHIDResource/IOHIDResourceDeviceUserClient/IOHIDUserDevice'", "'path': 'IOService:/IOResources/IOHIDResource/IOHIDResourceDeviceUserClient/IOHIDUserDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@0/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@2/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@2/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@2/AppleUserUSBHostHIDDevice'", "'path': 'IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/RP13@1D,4/IOPP/IOBC@0,1/IOBufferCopyController/AppleUSBVHCIBCE@80000000/AppleUSBVHCIPort@80500000/Apple Internal Keyboard / Trackpad@80500000/HID Relay@2/AppleUserUSBHostHIDDevice'"]
```
